src/js/views/maps/ViewfinderView.js

```diff
@@ -0,0 +1,154 @@
+     * @type {string}
+     */
+    type: "ViewfinderView",
+
+    /**
+     * The HTML classes to use for this view's element
+     * @type {string}
+     */
+    className: classNames.baseClass,
+
+    /**
+    * The events this view will listen to and the associated function to call.
+    * @type {Object}
+    */
+    events: {
+      [`change .${classNames.input}`]: 'valueChange',
+      [`click .${classNames.button}`]: 'search',
+      [`keyup .${classNames.input}`]: 'keyup',
+    },
+
+    /** 
+     * Values meant to be used by the rendered HTML template.
+     */
+    templateVars: {
+      errorMessage: "",
+      // Track the input value across re-renders.
+      inputValue: "",
+      placeholder: "Search by latitude and longitude",
+      classNames,
+    },
+
+    /**
+     * @typedef {Object} ViewfinderViewOptions
+     * @property {Map} The Map model associated with this view allowing control
+     * of panning to different locations on the map. 
+     */
+    initialize(options) {
+      this.mapModel = options.model;
+    },
+
+    render() {
+      this.el.innerHTML = _.template(Template)(this.templateVars);
+
+      this.focusInput();
+    },
+
+    focusInput() {
+      const input = this.getInput();
+      input.focus();
+      // Move cursor to end of input.
+      input.val("");
+      input.val(this.templateVars.inputValue);
+    },
+
+    getInput() {
+      return this.$el.find(`.${classNames.input}`);
+    },
+
+    getButton() {
+      return this.$el.find(`.${classNames.button}`);
+    },
+
+    /** Event handler for Backbone.View configuration. */
+    keyup(event) {
+      if (event.key === "Enter") {
+        this.search();
+      }
+    },
+
+    valueChange() {
+      this.templateVars.inputValue = this.getInput().val();
+    },
+
+    /** Event handler for Backbone.View configuration. */
+    search() {
+      this.clearError();
+
+      const coords = this.parseValue(this.templateVars.inputValue)
+      if (!coords) return;
+
+      this.model.zoomTo({ ...coords, height: 10000 /* meters */ });
+    },
+
+    /**
+     * Parse the user's input as a pair of floating point numbers. Log errors to the UI
+     * @return {{Number,Number}|undefined} Undefined represents an irrecoverable user input,
+     *   otherwise returns a latitude, longitude pair.
+     */
+    parseValue(value) {
+      const matches = value.match(floatsRegex);
+      const hasBannedChars = value.match(bannedCharactersRegex) != null;
+      if (matches?.length !== 2 || isNaN(matches[0]) || isNaN(matches[1]) || hasBannedChars) {
+        this.setError("Try entering a search query with two numerical values representing a latitude and longitude (e.g. 64.84, -147.72).");
+        return;
+      }
+
+      const latitude = Number(matches[0]);
+      const longitude = Number(matches[1]);
+      if (latitude > 90 || latitude < -90) {
+        this.setError("Latitude values outside of the range of -90 to 90 may behave unexpectedly.");
```

Comment:
Curious, how does it behave? Is the behaviour ever useful?

Reply:
It behaves appropriately as far as I can tell. I mean this error message mostly to make sure that the user meant to enter the value that they entered. For example if they accidentally type "47, -1701.2" instead of "47, -170.12", it's likely that they should double check their input even though the Cesium viewer will take them to -1701.2 mod 180

Reply:
Interesting! My initial expectation was for the viewfinder to show an error message and _not_ zoom when the coords are outside the valid range. Maybe we could ask the researchers & users if there's ever a case where they would want to enter outside the 90/-90 180/-180 range? Maybe we could indicate somehow the corrected lat,lon that the map actually navigates to in these cases?